### ISSUE TITLE
Lab Panels

### ISSUE THREAD

**Issue by arscan** [synthetichealth/synthea#369]
We are starting to include more 'lab panels' in our modules, which currently involve creating multiple `Observation` states (8 for one common lab panel), followed by a `DiagnosticReport` that references those `Observation`s.  If a module includes multiple lab panels, the module quickly becomes hard to read (and even harder to create).

There are a couple of options if we want to make this easier for users:

1) Improve the synthea module builder to make it easer to create and display these lab panels, automatically rolling DiagnosticReports up with their Observations when appropriate.  Pro: doesn't involve touching the synthea engine.  Con: It would be hard to implement well in the module builder, particularly on the 'read' side, and could potentially introduce confusion for the user to have some kind of 'roll up' state.
2) Add a dedicated new state type that represents these lab panels, which the synthea engine interprets as observations and diagnostic reports.  Pro: minimal updates needed to module builder, no changes needed for existing modules.  Con: update needed for synthea engine, documentation, and it seems to overlap heavily with the `DiagnosticReport` state conceptually.
3) Update `DiagnosticReport` to allow users to specify observations directly within, instead of having to create dedicated observation states.  We could have it support the user either specifying `last n` observations, or providers the observations directly within.  Pros: same as above, except we would need to update existing modules using `DiagnosticReport`.  Cons: same as above, except there would be less redundancy than if we created a dedicated type.

I recommend the third option, and I'm good with either changing it to only allow users to embed observations, or to give them the option of keeping the `last n` method.  The `last n` method does provide more flexibility (in case there are multiple paths of observations leading into a single `DiagnosticReport`), but I'm skeptical that level of flexibility is necessary.

@jawalonoski @dehall -- I spoke w/both of you about this, and it seems like the third option is ok.  Is there anything we missed that would point to one of the other solutions (or a different one than I listed?)

**Comment by dehall** [comment]
Assuming we go down the third path, in terms of implementation is there any reason we couldn't essentially embed Observation states within the DiagnosticReport? Everything except the `type` and `transition` fields is still relevant. That would make the implementation in the engine almost trivial. But I'm not sure whether that would work great in the UI or solve the UX issue.

Ex, something like this from wellness_encounters:


Before:
```
    "Record_Cholesterol": {
      "type": "Observation",
      "target_encounter": "Wellness_Encounter",
      "category": "laboratory",
      "vital_sign": "Total Cholesterol",
      "codes": [
        {
          "system": "LOINC",
          "code": "2093-3",
          "display": "Total Cholesterol"
        }
      ],
      "unit": "mg/dL",
      "direct_transition": "Record_Triglycerides"
    },
    "Record_Triglycerides": {
      "type": "Observation",
      "target_encounter": "Wellness_Encounter",
      "category": "laboratory",
      "vital_sign": "Triglycerides",
      "codes": [
        {
          "system": "LOINC",
          "code": "2571-8",
          "display": "Triglycerides"
        }
      ],
      "unit": "mg/dL",
      "direct_transition": "Record_LDL"
    },
    "Record_LDL": {
      "type": "Observation",
      "target_encounter": "Wellness_Encounter",
      "category": "laboratory",
      "vital_sign": "LDL",
      "codes": [
        {
          "system": "LOINC",
          "code": "18262-6",
          "display": "Low Density Lipoprotein Cholesterol"
        }
      ],
      "unit": "mg/dL",
      "direct_transition": "Record_HDL"
    },
    "Record_HDL": {
      "type": "Observation",
      "target_encounter": "Wellness_Encounter",
      "category": "laboratory",
      "vital_sign": "HDL",
      "codes": [
        {
          "system": "LOINC",
          "code": "2085-9",
          "display": "High Density Lipoprotein Cholesterol"
        }
      ],
      "unit": "mg/dL",
      "direct_transition": "Record_LipidPanel"
    },
    "Record_LipidPanel": {
      "type": "DiagnosticReport",
      "number_of_observations": 4,
      "codes": [
        {
          "system": "LOINC",
          "code": "57698-3",
          "display": "Lipid Panel"
        }
      ],
      "target_encounter": "Wellness_Encounter",
      "direct_transition": "Lab_ACR"
    },
```

After:
```
    "Record_LipidPanel": {
      "type": "DiagnosticReport",
      "number_of_observations": 4,
      "codes": [
        {
          "system": "LOINC",
          "code": "57698-3",
          "display": "Lipid Panel"
        }
      ],
      "observations": [
        {
          "category": "laboratory",
          "vital_sign": "Total Cholesterol",
          "codes": [
            {
              "system": "LOINC",
              "code": "2093-3",
              "display": "Total Cholesterol"
            }
          ],
          "unit": "mg/dL"
        },
        {
          "category": "laboratory",
          "vital_sign": "Triglycerides",
          "codes": [
            {
              "system": "LOINC",
              "code": "2571-8",
              "display": "Triglycerides"
            }
          ],
          "unit": "mg/dL"
        },
        {
          "category": "laboratory",
          "vital_sign": "LDL",
          "codes": [
            {
              "system": "LOINC",
              "code": "18262-6",
              "display": "Low Density Lipoprotein Cholesterol"
            }
          ],
          "unit": "mg/dL"
        },
        {
          "category": "laboratory",
          "vital_sign": "HDL",
          "codes": [
            {
              "system": "LOINC",
              "code": "2085-9",
              "display": "High Density Lipoprotein Cholesterol"
            }
          ],
          "unit": "mg/dL"          
        }
      ],
      "target_encounter": "Wellness_Encounter",
      "direct_transition": "Lab_ACR"
    },
```

Maybe we'd want to allow, not necessarily require, a common `category` and `unit` as well.

**Comment by jawalonoski** [comment]
Well, that seems reasonable... One observation about @dehall's suggestion -- that solution only works on Lab Panels. A user will still have to create lots of observations states if they aren't grouped in a panel. That might be acceptable, possibly preferred (?), but we need to acknowledge that.

**Comment by dehall** [comment]
Implemented in #371
